airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/BasicAcceptanceTests.java

```diff
@@ -712,7 +712,160 @@ public void testResetCancelsRunningSync() throws Exception {
     assertEquals(JobStatus.CANCELLED, connectionSyncReadAfterReset.getStatus());
   }
 
-  // This test is disabled because it takes a couple minutes to run, as it is testing timeouts.
+  @Test
+  public void testSyncAfterUpgradeToPerStreamState(final TestInfo testInfo) throws Exception {
+    LOGGER.info("Starting {}", testInfo.getDisplayName());
+    final String connectionName = "test-connection";
+    final SourceRead source = testHarness.createPostgresSource();
+    final UUID sourceId = source.getSourceId();
+    final UUID sourceDefinitionId = source.getSourceDefinitionId();
+    final UUID destinationId = testHarness.createDestination().getDestinationId();
+    final UUID operationId = testHarness.createOperation().getOperationId();
+    final AirbyteCatalog catalog = testHarness.discoverSourceSchema(sourceId);
+    final AirbyteStream stream = catalog.getStreams().get(0).getStream();
+
+    assertEquals(Lists.newArrayList(SyncMode.FULL_REFRESH, SyncMode.INCREMENTAL), stream.getSupportedSyncModes());
+    // instead of assertFalse to avoid NPE from unboxed.
+    assertNull(stream.getSourceDefinedCursor());
+    assertTrue(stream.getDefaultCursorField().isEmpty());
+    assertTrue(stream.getSourceDefinedPrimaryKey().isEmpty());
+
+    // Set the source to a version that does not support per-stream state
+    LOGGER.info("Setting source connector to pre-per-stream state version {}...",
+        AirbyteAcceptanceTestHarness.POSTGRES_SOURCE_LEGACY_CONNECTOR_VERSION);
+    testHarness.updateSourceDefinitionVersion(sourceDefinitionId, AirbyteAcceptanceTestHarness.POSTGRES_SOURCE_LEGACY_CONNECTOR_VERSION);
+
+    final SyncMode syncMode = SyncMode.INCREMENTAL;
+    final DestinationSyncMode destinationSyncMode = DestinationSyncMode.APPEND;
```

Comment:
VERY nit: Is this representative of most of our connection I think that we could use the default syncMode which is the `APPEND_DEDUP` one. I don't know the implementation details of it but it might be an extension of the `APPEND` so having it will test the APPEND and the DEDUP.

Reply:
Done